process/interceptors/processor/trieNodeChunksProcessor.go

```diff
@@ -1,3 +1,214 @@
+type TrieNodesChunksProcessorArgs struct {
+	Hasher          hashing.Hasher
+	ChunksCacher    storage.Cacher
+	RequestInterval time.Duration
+	RequestHandler  process.RequestHandler
+	Topic           string
+	ShardID         uint32
+}
+
+type trieNodeChunksProcessor struct {
+	hasher            hashing.Hasher
+	chunksCacher      storage.Cacher
+	chanCheckRequests chan checkRequest
+	requestInterval   time.Duration
+	requestHandler    process.RequestHandler
+	topic             string
+	shardID           uint32
+	cancel            func()
+}
+
+// NewTrieNodeChunksProcessor creates a new trieNodeChunksProcessor instance
+func NewTrieNodeChunksProcessor(arg TrieNodesChunksProcessorArgs) (*trieNodeChunksProcessor, error) {
+	if check.IfNil(arg.Hasher) {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrNilHasher)
+	}
+	if check.IfNil(arg.ChunksCacher) {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrNilCacher)
+	}
+	if arg.RequestInterval < minimumRequestTimeInterval {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor, minimum request interval is %v",
+			process.ErrInvalidValue, minimumRequestTimeInterval)
+	}
+	if check.IfNil(arg.RequestHandler) {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrNilRequestHandler)
+	}
+	if len(arg.Topic) == 0 {
+		return nil, fmt.Errorf("%w in NewTrieNodeChunksProcessor", process.ErrEmptyTopic)
+	}
+
+	tncp := &trieNodeChunksProcessor{
+		hasher:            arg.Hasher,
+		chunksCacher:      arg.ChunksCacher,
+		chanCheckRequests: make(chan checkRequest),
+		requestInterval:   arg.RequestInterval,
+		requestHandler:    arg.RequestHandler,
+		topic:             arg.Topic,
+		shardID:           arg.ShardID,
+	}
+	var ctx context.Context
+	ctx, tncp.cancel = context.WithCancel(context.Background())
+	go tncp.processLoop(ctx)
+
+	return tncp, nil
+}
+
+func (proc *trieNodeChunksProcessor) processLoop(ctx context.Context) {
+	chanDoRequests := time.After(proc.requestInterval)
+	for {
+		select {
+		case <-ctx.Done():
+			log.Debug("trieNodeChunksProcessor.processLoop go routine is stopping...")
+			return
+		case request := <-proc.chanCheckRequests:
+			proc.processCheckRequest(request)
+		case <-chanDoRequests:
+			proc.doRequests(ctx)
+			chanDoRequests = time.After(proc.requestInterval)
+		}
+	}
+}
+
+// CheckBatch will check the batch returning a checked chunk result containing result processing
+func (proc *trieNodeChunksProcessor) CheckBatch(b *batch.Batch) (process.CheckedChunkResult, error) {
+	batchValid, err := proc.batchIsValid(b)
+	if !batchValid {
+		return process.CheckedChunkResult{
+			IsChunk:        false,
+			HaveAllChunks:  false,
+			CompleteBuffer: nil,
+		}, err
+	}
+
+	respChan := make(chan process.CheckedChunkResult, 1)
+	req := checkRequest{
+		batch:        b,
+		chanResponse: respChan,
+	}
+
+	proc.chanCheckRequests <- req
+	response := <-respChan
```

Comment:
add a select here as to not wait indefinitely if the context is done and the respChan will never hold a value.

Reply:
done, thanks